### ISSUE TITLE
[data grid] Column header menu button aria-labels should be descriptive and unique

### ISSUE THREAD

**Issue by dil-pvajda** [mui/mui-x#16454]
Relates to #16456

### Steps to reproduce

Steps:
1. Open any MUI DataGrid example, eg.: https://mui.com/x/react-data-grid/column-definition/
2. Inspect the menu button inside column headers with DevTools

### Current behavior

- All the menu buttons have identical labels, "Menu", which makes them hard to tell apart from each other.
- Because of `aria-haspopup`, the menu button will be announced to a screen reader user as "Menu menu button", which is inappropriate.
- Screen reader users will be unable to determine the purpose of these buttons

### Expected behavior

Ideally we'd be able to customize the label that is used for each column, possibly adding the name of the column and more descriptive text about the purpose of the button.

The default localization of the component should probably be revisited to resolve the underlying issue.

### Context

We're in an accessibility audit with our design system and we cannot resolve these issues.
Originally identified by Level Access.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: macOS 15.3
  Binaries:
    Node: 20.18.0 - ~/.local/share/nvm/v20.18.0/bin/node
    npm: 10.8.2 - ~/.local/share/nvm/v20.18.0/bin/npm
    pnpm: 10.1.0 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: Not Found
    Edge: Not Found
    Safari: 18.3
  npmPackages:
    @emotion/react: 11.13.5 => 11.13.5 
    @mui/icons-material: ^6.1.9 => 6.1.9 
    @mui/lab: 6.0.0-beta.25 => 6.0.0-beta.25 
    @mui/material: ^6.4.2 => 6.4.2 
    @mui/system: ^6.4.2 => 6.4.2 
    @mui/utils: ^6.4.2 => 6.4.2 
    @mui/x-data-grid-pro: ^7.25.0 => 7.25.0 
    @mui/x-date-pickers: ^7.25.0 => 7.25.0 
    @mui/x-date-pickers-pro: ^7.25.0 => 7.25.0 
    @types/react: ^18.2.51 => 18.3.12 
    react: ^18.2.0 => 18.3.1 
    react-dom: ^18.2.0 => 18.3.1 
    typescript: ^4.8.4 => 4.9.5 
```
</details>

**Search keywords**: accessibility, a11y, column header, menu button

**Order ID**: 75534

**Comment by KenanYusuf** [comment]
Hi @dil-pvajda — thanks for opening the issue.

Do you think this would be a better format for the column menu button `aria-label`?

`{columnName} column options`

Any suggestions would be welcome.

**Comment by dil-pvajda** [comment]
Hi @KenanYusuf, Yes, I think your suggestion would work well, and it would also combine with the default announcement of "menu".

**Comment by michelengelen** [comment]
All right ... I'll add this to the board and additionally open it up as a "good first issue"

Here is a diff to get started:
```diff
--- a/packages/x-data-grid/src/components/columnHeaders/ColumnHeaderMenuIcon.tsx
+++ b/packages/x-data-grid/src/components/columnHeaders/ColumnHeaderMenuIcon.tsx
@@ -45,10 +45,12 @@ export const ColumnHeaderMenuIcon = React.memo((props: ColumnHeaderMenuIconProps
     [apiRef, colDef.field],
   );

+  const ariaLabel = `${colDef.headerName || colDef.field} ${apiRef.current.getLocaleText('columnMenuLabel')}`;
+
   return (
     <div className={classes.root}>
       <rootProps.slots.baseTooltip
-        title={apiRef.current.getLocaleText('columnMenuLabel')}
+        title={ariaLabel}
         enterDelay={1000}
         {...rootProps.slotProps?.baseTooltip}
       >
@@ -56,7 +58,7 @@ export const ColumnHeaderMenuIcon = React.memo((props: ColumnHeaderMenuIconProps
           ref={iconButtonRef}
           tabIndex={-1}
           className={classes.button}
-          aria-label={apiRef.current.getLocaleText('columnMenuLabel')}
+          aria-label={ariaLabel}
           size="small"
           onClick={handleMenuIconClick}
           aria-haspopup="menu"
diff --git a/packages/x-data-grid/src/constants/localeTextConstants.ts b/packages/x-data-grid/src/constants/localeTextConstants.ts
index 5f528025c..731187032 100644
--- a/packages/x-data-grid/src/constants/localeTextConstants.ts
+++ b/packages/x-data-grid/src/constants/localeTextConstants.ts
@@ -120,7 +120,7 @@ export const GRID_DEFAULT_LOCALE_TEXT: GridLocaleText = {
   filterValueFalse: 'false',

   // Column menu text
-  columnMenuLabel: 'Menu',
+  columnMenuLabel: 'column options',
   columnMenuShowColumns: 'Show columns',
   columnMenuManageColumns: 'Manage columns',
   columnMenuFilter: 'Filter',
```

This would be the first start and the other languages localeTexts would need to be invalidated.

cc @KenanYusuf
